src/BuildingRegistry/Building/Building.cs

```diff
@@ -41,7 +42,7 @@ public sealed partial class Building : AggregateRootEntity, ISnapshotable
         {
             var geometry = WKBReaderFactory.Create().Read(extendedWkbGeometry);
 
-            GuardPolygon(geometry);
+            GuardPolygon(geometry, isMeasured: false);
```

Comment:
Plan was using the `IsSelfTouchingRingFormingHoleValid = true` thefore it didn't throw an exception with invalid selftouching polygon